task/sast-unicode-check/0.1/sast-unicode-check.yaml

```diff
@@ -0,0 +1,265 @@
+            fi
+
+            # Check usage of find-unicode-control to confirm the tool can work
+            ./cloned/find_unicode_control.py --help >&2
+            status=$?
+            if [ "${status}" -ne 0 ]; then
+                echo "Error: Failed to check usage of find-unicode-control" >&2
+                return 1
+            else
+                echo "Message: Succeed to check usage of find-unicode-control" >&2
+            fi
+        }
+
+        # Find unicode control
+        unicontrol_scan()
+        {
+            LANG=en_US.utf8 ./cloned/find_unicode_control.py -p bidi -v -d -t "${SOURCE_CODE_DIR}/source" \
+                >raw_sast_unicode_check_out.txt \
+                2>raw_sast_unicode_check_out.log \
+                || FUC_EXIT_CODE=$?
+
+            if [[ -z "${FUC_EXIT_CODE}" || "${FUC_EXIT_CODE}" -eq 1 ]]; then
+                # No non-printable unicode characters found, or successfully found
+                echo "Message: Succeed to run find-unicode-control command" >&2
+            else
+                # Failed to run the command
+                echo "Error: failed to run find unicode control command" >&2
+                return 1
+            fi
+
+            # Translate the output format
+            sed -i raw_sast_unicode_check_out.txt -E -e 's|(.*:[0-9]+)(.*)|\1: warning:\2|' -e 's|^|Error: UNICONTROL_WARNING:\n|'
+            status=$?
+            if [ "${status}" -ne 0 ] && [ "${status}" -ne 1 ]; then
+                echo "Error: failed to translate the unicontrol output format" >&2
+                return 1
+            else
+                echo "Message: Succeed to translate the unicontrol output format" >&2
+            fi
+        }
+
+        # Process all results as configured with CSGERP_OPTS
+        process_all_results()
+        {
+            CSGERP_OPTS=(
+                --mode=json
+                --event='error|warning'
+                --remove-duplicates
+                --embed-context=3
+                --set-scan-prop="${SCAN_PROP}"
+            )
+            # In order to generate csdiff/v1, we need to add the whole path of the source code as Snyk only provides an URI to embed the context
+            csgrep --mode=json --prepend-path-prefix="${SOURCE_CODE_DIR}"/ raw_sast_unicode_check_out.txt \
+                | csgrep "${CSGERP_OPTS[@]}" \
+                | csgrep --mode=json --strip-path-prefix="${SOURCE_CODE_DIR}"/source/ \
+                > processed_sast_unicode_check_out.json
+
+            csgrep --mode=evtstat processed_sast_unicode_check_out.json
+        }
+
+        # Filter known false positive
+        filter_kfp()
+        {
+            # We check if the KFP_GIT_URL variable is set to apply the filters or not
+            if [[ -z "${KFP_GIT_URL}" ]]; then
+                mv processed_sast_unicode_check_out.json sast_unicode_check_out.json
+                echo "Error: KFP_GIT_URL variable not defined. False positives won't be filtered" >&2
```

Comment:
From end user perspective, `KFP_GIT_URL` is a task parameter.  That is how we document it.

Reply:
Updated, remove the judgment, only use the `KFP_GIT_URL ` exist logic